org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/PDEMavenBundlePluginConfigurator.java

```diff
@@ -0,0 +1,174 @@
+
+/**
+ * This configurator performs the following tasks:
+ * <ul>
+ * <li>Enable the PDE nature for this project to make PDE aware of this
+ * project</li>
+ * <li>Set the location of the "bundle-root" where PDE looks for the
+ * manifest</li>
+ * </ul>
+ */
+@SuppressWarnings("restriction")
+public class PDEMavenBundlePluginConfigurator extends AbstractProjectConfigurator implements IJavaProjectConfigurator {
+
+	private static final String FELIX_PARAM_MANIFESTLOCATION = "manifestLocation";
+	private static final String FELIX_PARAM_SUPPORTINCREMENTALBUILD = "supportIncrementalBuild";
+	private static final String FELIX_MANIFEST_GOAL = "manifest";
+	private static final String BND_PARAM_MANIFESTLOCATION = "manifestPath";
+	private static final String[] BND_MANIFEST_GOALS = { "bnd-process", "bnd-process-tests", "jar", "test-jar" };
+
+	@Override
+	public void configure(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+		List<MojoExecution> executions = getMojoExecutions(request, monitor);
+		List<MavenProblemInfo> problems = new ArrayList<MavenProblemInfo>();
+		boolean hasManifestExecution = false;
+		for (MojoExecution execution : executions) {
+			Plugin plugin = execution.getPlugin();
+			if (isFelix(plugin)) {
+				MavenProject mavenProject = request.getMavenProject();
+				IMaven maven = MavenPlugin.getMaven();
+				if (isFelixBundleGoal(execution)) {
+					Boolean supportIncremental = maven.getMojoParameterValue(mavenProject, execution,
+							FELIX_PARAM_SUPPORTINCREMENTALBUILD, Boolean.class, monitor);
+					if (supportIncremental == null || !supportIncremental.booleanValue()) {
+						SourceLocation location = SourceLocationHelper.findLocation(execution.getPlugin(),
+								"configuration");
+						MavenProblemInfo problem = new MavenProblemInfo(
+								"Incremental updates are currently disabled, set supportIncrementalBuild=true to support automatic manifest updates for this project.",
+								IMarker.SEVERITY_WARNING, location);
+						problems.add(problem);
+					}
+					hasManifestExecution = true;
+				}
+			} else if (isBND(plugin)) {
+				if (isBNDBundleGoal(execution)) {
+					hasManifestExecution = true;
+				}
+			}
+		}
+		if (!hasManifestExecution) {
+			for (MojoExecution execution : executions) {
+				SourceLocation location = SourceLocationHelper.findLocation(execution.getPlugin(), "executions");
+				MavenProblemInfo problem = new MavenProblemInfo(
+						"There is currently no execution that generates a manifest, consider adding an execution for one of the following goal: "
+								+ (isFelix(execution.getPlugin()) ? FELIX_MANIFEST_GOAL
+										: Arrays.toString(BND_MANIFEST_GOALS))
+								+ ".",
+						IMarker.SEVERITY_WARNING, location);
+				problems.add(problem);
+				break;
+			}
+		}
+		if (problems.size() > 0) {
+			this.markerManager.addErrorMarkers(request.getPom(), IMavenConstants.MARKER_LIFECYCLEMAPPING_ID, problems);
+		}
```

Comment:
This code does not consider that the resource of a problem could be the parent pom (or grand-parent etc.).
The following suggestion does consider this.
```suggestion
		if (!problems.isEmpty()) {
			IMavenProjectRegistry projectManager = MavenPlugin.getMavenProjectRegistry();
			for (MavenProblemInfo problem : problems) {
				String[] gav = problem.getLocation().getResourceId().split(":");
				IFile pom = projectManager.getMavenProject(gav[0], gav[1], gav[2]).getPom();
				markerManager.addErrorMarker(pom, IMavenConstants.MARKER_LIFECYCLEMAPPING_ID, problem);
			}
		}
```